### PR TITLE
Dataset Stash Unit Tests

### PR DESCRIPTION
**Pull request by letv3**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by letv3** [comment]
Marked `test_dataset_update` and `test_dataset_serach_action_ids` as **xfail** because it seams some corresponding functionality is not implemented.

**Comment by letv3** [comment]
Hey @shubham3121 @koenvanderveen  Can you please review the PR? 
Thanks

File: packages/syft/tests/syft/dataset/dataset_stash_test.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from syft.core.node.new.dataset import Dataset
+from syft.core.node.new.dataset_stash import ActionIDsPartitionKey
+from syft.core.node.new.dataset_stash import NamePartitionKey
+from syft.core.node.new.document_store import QueryKey
+from syft.core.node.new.document_store import QueryKeys
+from syft.core.node.new.uid import UID
+
+
+def test_dataset_namepartitionkey():
+    name_partition_key = QueryKeys(qks=NamePartitionKey).all[0]
```
Comment (shubham3121) [inline packages/syft/tests/syft/dataset/dataset_stash_test.py:17]: To generate a Querykey from PartitionKey we can either:
`QueryKey.from_obj(partition_key=NamePartitionKey, obj='Dummy Name')`
or call
name_query_key = NamePartitionKey.with_obj('Dummy Name')  # pass in the value directly for the partition key
Or
@dataclass
class MockObj:
    name: str

obj = MockObj(name="Dummy Name")
name_query_key = NamePartitionKey.with_obj(obj)  # pass an object which has partition key as one of its class members.

Note: A QueryKey is has the attributes of the PartitionKey plus `value`, which holds the value.

You can refer to this test for [reference](https://github.com/OpenMined/PySyft/pull/7462/files#diff-39617657524bd6b2b22dfb196f13b928a252c0f4fd2ed76e5584bf7c85d996f3R82)

File: packages/syft/tests/syft/dataset/dataset_stash_test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    assert result.err(), (
+        f"Dataset was updated with non-DatasetUpdate object," f"result: {result}"
+    )
+
+
+@pytest.mark.xfail(
+    raises=(NotImplementedError, AssertionError),
+    reason="StorePartition.find_index_or_search_keys is not implemeted yet",
+)
+def test_dataset_serach_action_ids(mock_dataset_stash, mock_dataset):
```
[the remark this answers is not in the document]
Comment (shubham3121) [reply packages/syft/tests/syft/dataset/dataset_stash_test.py:64]: A Dataset is composed of a list of assets (see Asset class). This asset actual actually stores the data. `action_id` is the location of the asset in the ActionStore.

For reference:
https://github.com/OpenMined/PySyft/blob/dev/notebooks/api/0.8/00-load-data.ipynb

So to test this, we need to:
- create a asset and save it to ActionStore
- create a Dataset with Asset attached to it.
- save Dataset to store.
Now we can retrieve datasets for given action ids.

**Review by shubham3121 — CHANGES_REQUESTED** [review]
@letv3 left some comments. Feel free to reach out to me for any clarifications.

**Review by madhavajay — APPROVED** [review]
This is awesome work @letv3! 🙏

**Pull request merged**